src/uint/sqrt.rs

```diff
@@ -1,17 +1,36 @@
 //! [`Uint`] square root operations.
 
 use super::Uint;
-use crate::{Limb, Word};
 use subtle::{ConstantTimeEq, CtOption};
 
 impl<const LIMBS: usize> Uint<LIMBS> {
-    /// See [`Self::sqrt_vartime`].
-    #[deprecated(
-        since = "0.5.3",
-        note = "This functionality will be moved to `sqrt_vartime` in a future release."
-    )]
-    pub const fn sqrt(&self) -> Self {
-        self.sqrt_vartime()
+    /// Computes √(`self`) in constant time.
+    /// Based on Brent & Zimmermann, Modern Computer Arithmetic, v0.5.9, Algorithm 1.13
+    ///
+    /// Callers can check if `self` is a square by squaring the result
+    pub fn sqrt(&self) -> Self {
+        let max_bits = (self.bits() + 1) >> 1;
+        let cap = Self::ONE.shl(max_bits);
+        let mut guess = cap; // ≥ √(`self`)
+        let mut xn = {
+            let q = self.wrapping_div(&guess);
```

Comment:
My mistake, I didn't see that `wrapping_div` is only constant-time for a fixed `rhs`. Actually, I think there might be a documentation issue here—I can't find anywhere in the public documentation that says this, only the documentation for `ct_div_rem` (which doesn't appear in the public docs since it's `pub(crate)`).

What do you think would be the better approach here: making a new function that's like `ct_div_rem` but constant-time with respect to both inputs, or modifying `ct_div_rem` to have this stronger constant-time guarantee and moving the "constant-time only for fixed rhs" behavior to a new function?

Reply:
I agree that the name is misleading - it should have had a `vartime` suffix (and the whole `div.rs` is kind of a mess in terms of naming - see #268). So the proper way to proceed I think would be to rename the current one to `_vartime`, and implement a constant-time one in its place.

Reply:
It appears this is a blocker on merging this PR. I guess we can go ahead and flip over to the v0.6 series per #268 and try to land this PR afterward.

Reply:
Checking on this: `wrapping_div` calls `const_div_rem`, which claims:

```rust
    /// This function is constant-time with respect to both `self` and `rhs`.
    pub(crate) const fn const_div_rem(&self, rhs: &Self) -> (Self, Self, CtChoice) {
```

Reply:
Which commit are you looking at? The current `master` has no `const_div_rem()`, and `Uint::wrapping_div()` uses `ct_div_rem()`, which is not constant-time in `rhs`.

Reply:
This PR changes `wrapping_div` to call `const_div_rem`, which is also added by this PR